fuzz/do_fuzz.py

```diff
@@ -0,0 +1,129 @@
+import json
+import logging
+import os
+import sys
+import random
+
+
+def generate_data(template):
+    with open(template) as f:
+        template_file = json.load(f)
+    for obj in template_file:
+        data_var = template_file[obj]['data']
+        format_var = template_file[obj]['format']
+        if isinstance(data_var, list):
+            result = revised_generate_list(data_var, format_var)
+            construct_json(result, obj, template)
+        elif isinstance(data_var, int):
+            result = generate_int(format_var)
+            construct_json(result, obj, template)
+        else:
+            raise TypeError
+
+
+def revised_generate_list(input_data, data_format):
+    try:
+        # base case
+        if isinstance(input_data, int):
+            return generate_int(data_format)
+        else:
+            # recurrence
+            for i in range(len(input_data)):
+                input_data[i] = revised_generate_list(input_data[i], data_format)
+        return input_data
+    except Exception:
+        raise TypeError
+
+
+def generate_int(data_format):
+    is_signed = data_format['is_signed']
+    width = data_format['width']
+
+    if is_signed:
+        result = random.randrange(-2 ** (width - 1) + 1, 2 ** (width - 1) - 1)
+    else:
+        result = random.randrange(0, 2 ** width - 1)
+    return result
+
+
+def construct_json(data_list, obj, template):
+    with open(template) as json_file:
+        data = json.load(json_file)
+    data[obj]["data"] = data_list
+
+    with open(template, "w") as json_file:
+        json.dump(data, json_file)
+
+
+def compare_object(a, b):
+    if type(a) != type(b):
+        return False
+    elif type(a) is dict:
+        return compare_dict(a, b)
+    elif type(a) is list:
+        return compare_list(a, b)
+    else:
+        return a == b
```

Comment:
Why are the hand-crafted comparisons necessary?  Are the built in comparison operators for list and dictionary not sufficient?

Reply:
Let me know if my question here is unclear. This is the only comment I have left.